### ISSUE TITLE
VSG gives error when setting names in a different case than the prefix

### ISSUE THREAD

**Issue by rugebiker** [jeremiah-c-leary/vhdl-style-guide#644]
**Environment**
VSG 3.3.0. Ubuntu 20.04

**Describe the bug**

I think this bug would affect all generics, constants, etc, and for both prefix and suffix, but for the example I will use "generic" rules and prefix:

- Rule "generic_007" defines that generic names have the proper case.
- Rule "generic_020" defines a prefix for generics.

If I set the following rules:

```
{
  "generic_007": {
      "case" : "upper"
  },
  "generic_020": {
      "prefixes": ["g_"]
  }
}
```
(so enforcing upper case, with prefix g_ [note that g_ is lower case])

and I have a generic in my VHDL code like this:

`g_THIS_IS_A_GENERIC`

VSG gives me the error that g_ should also be capital.

**Expected behavior**
I would expect that the generic name case rule only applies to the generic name, and that the prefix would be respected as it is exactly defined. So in the example above, it shouldn't give an error.

Thanks a lot :)
